hyperstyle/src/python/review/inspectors/pmd/pmd.py

```diff
@@ -28,26 +25,30 @@
 
 class PMDInspector(BaseInspector):
     inspector_type = InspectorType.PMD
+    has_access = False
 
-    def __init__(self):
-        os.chmod(PATH_TOOLS_PMD_SHELL_SCRIPT, 0o777)
-
-    @classmethod
-    def _create_command(cls, path: Path,
+    def _create_command(self, path: Path,
                         output_path: Path,
                         language_version: LanguageVersion,
                         n_cpu: int) -> List[str]:
+        path_tools_pmd_shell = f'{os.environ[PMD_DIRECTORY_ENV]}/pmd-bin-{os.environ[PMD_VERSION_ENV]}/bin/run.sh'
+        if not self.has_access:
+            os.chmod(path_tools_pmd_shell, 0o777)
```

Comment:
`os.chmod` may throw `PermissionError`, you can also use `os.access(path_tools_pmd_shell, os.X_OK)` instead of `self.access`. Although you can leave it as it is.